online-shop/src/components/Product.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import TableRow from '@material-ui/core/TableRow';
+import TableBody from '@material-ui/core/TableBody';
+import Button from '@material-ui/core/Button';
+import {Route, Link, Switch} from 'react-router-dom';
+import ProductDetails from './ProductDetails';
+import { Theme } from '@material-ui/core/styles/createMuiTheme';
+import withStyles from '@material-ui/core/styles/withStyles';
+import createStyles from '@material-ui/core/styles/createStyles';
+import TableCell from '@material-ui/core/TableCell/TableCell';
+
+const MyCellType = withStyles((theme: Theme) =>
+  createStyles({
+    body: {
+      fontSize: 25,
+    }
+  }),
+)(TableCell);
+
+function Product(props: { id: string, name: string; category: string, price: number, details: string; }) {
```

Comment:
Vad ca mai sus ai folosit si arrow functions pentru componente. Opteaza pentru una din variante, nu le alterna, ca te complici cand / daca o sa folosesti 'this' in vreuna. As sugera arrow functions.